### ISSUE TITLE
`sentry-delayed_job` causing issues with `Delayed::Job` and mocks in RSpec tests

### ISSUE THREAD

**Issue by TomasBarry** [getsentry/sentry-ruby#1334]
**Describe the bug**

Including `sentry-delayed_job` results in RSpec tests that make use of `Delayed::Job` and mocks to fail.

Could this perhaps be a clashing namespace issue or is the `sentry-delayed_job` middleware interfering with RSpec mocks?

**To Reproduce**

Run the following test with and without `sentry-delayed_job` in the Gemfile:

```ruby
require 'rails_helper'

RSpec.describe do
  class Thing
    def self.invoked_method; end
  end

  it 'example test' do
    allow(Thing).to receive(:invoked_method)
    expect(Delayed::Job.count).to eq(0)

    Thing.delay.invoked_method
    expect(Delayed::Job.count).to eq(1)

    Delayed::Worker.new.run(Delayed::Job.last)
    expect(Delayed::Job.count).to eq(0)

    expect(Thing)
      .to have_received(:invoked_method)
      .once
  end
end
```

A solution is to change the `Delayed::Worker.new.run(Delayed::Job.last)` line to:

```ruby
Delayed::Job.last.payload_object.perform
Delayed::Job.last.destroy
```

However, it would be preferable not to have to change our test suite and I am confused as to why the inclusion of `sentry-delayed_job` would require that change.

**Expected behavior**

The test should pass in both instances.

**Actual behavior**

The test passes without `sentry-delayed_job` in the Gemfile but fails when `sentry-delayed_job` is in the Gemfile with the following message:

```
Failures:

  1) example test
     Failure/Error:
       expect(Thing)
         .to have_received(:invoked_method)
         .once

       (Thing (class)).invoked_method(*(any args))
           expected: 1 time with any arguments
           received: 0 times with any arguments
```

**Environment**
 - Ruby Version: ruby-2.6.1 [ x86_64 ]
 - SDK Version:
```
sentry-delayed_job (4.2.1)
  sentry-ruby-core (~> 4.2.0)
sentry-raven (2.13.0)
  faraday (>= 0.7.6, < 1.0)
sentry-ruby-core (4.2.2)
  concurrent-ruby
  faraday
```
 - Integration Versions (if any):
  - Rails 6.0.3.5

**Comment by st0012** [comment]
@TomasBarry thanks for reporting, I'm investigating this right now. but I also notice that you're using `sentry-raven` with `sentry-ruby` together. those 2 are not compatible, can you replace `sentry-raven` with `sentry-ruby` first?

**Comment by chidicfc** [comment]
@st0012 It is the same with the `sentry-ruby` gem: it's the same behaviour.

**Comment by TomasBarry** [comment]
@st0012, I can confirm that the test still fails with `sentry-raven` removed.

**Comment by st0012** [comment]
thanks for all the feedback. I've fixed it and released it in version `4.3.1` now 🙂
